packages/dd-trace/src/plugins/util/web.js

```diff
@@ -427,11 +427,13 @@ function addRequestTags (context) {
   const { req, span, config } = context
   const url = extractURL(req)
 
+  const tags = span.context()._tags
+
   span.addTags({
     [HTTP_URL]: web.obfuscateQs(config, url),
     [HTTP_METHOD]: req.method,
     [SPAN_KIND]: SERVER,
-    [SPAN_TYPE]: WEB,
+    [SPAN_TYPE]: tags[SPAN_TYPE] === 'serverless' ? 'serverless' : WEB,
```

Comment:
This seems unfortunate. Should there be a `packages/dd-trace/src/plugins/util/serverless.js` file instead?

Reply:
Why was this even changed at all for Serverless? It seems like the span type is one of the oldest properties of a span with a special meaning dating back to the very beginning of APM. The Serverless backend should respect those semantics and add a property if it needs more information.

Reply:
We've been setting the span type to `serverless` on AWS Lambda functions for some time now. However this is currently happening in the Lambda layer. Azure Functions don't have the same concept of a layer where dependencies can be added so that's why it's being added as a plugin directly in the tracer.

https://github.com/DataDog/datadog-lambda-js/blob/e84f01785b2e761eba6f91e44dc277c1e7a78881/src/trace/listener.ts#L295-L296

Reply:
I added a new `SERVERLESS` property and also moved this type logic to `util/serverless.js`. I also needed to copy a few of the helper methods over from `util/web.js` to get this working. Let me know if you had another solution in mind.

https://github.com/DataDog/dd-trace-js/blob/bcd230e7ed58786c70df4d08d1f6700ed7f2eceb/ext/types.js#L5

Reply:
Could `serverless.js` just import web.js instead and only override the span type?

Reply:
Since the `addRequestTags` helper function isn't exported, I needed to pass span type to it as a parameter. Now `serverless.js` just imports `web.js` and overrides the `SPAN_TYPE` property I added.

https://github.com/DataDog/dd-trace-js/blob/ebbe75e66045a3558a11be8ca58376749d6f881c/packages/dd-trace/src/plugins/util/serverless.js#L1-L7

 I think this is the best approach for making `serverless.js` as simple as possible with minimal changes to `web.js`. Let me know if you think otherwise or have more suggestions for improving the code.